src/internal_events/common.rs

```diff
@@ -120,11 +120,22 @@ pub struct ComponentEventsDropped {
 
 impl InternalEvent for ComponentEventsDropped {
     fn emit(self) {
-        error!(
-            message = "Events dropped.",
-            intentional = self.intentional,
-            reason = self.reason,
-        );
+        let message = "Events dropped";
+        if self.intentional {
+            debug!(
+                message,
+                intentional = self.intentional,
+                reason = self.reason,
+                internal_log_rate_secs = 10,
+            );
+        } else {
+            error!(
+                message,
+                intentional = self.intentional,
+                reason = self.reason,
+                internal_log_rate_secs = 10,
+            );
+        }
```

Comment:
`self.intentional` could arguably be a `const INTENTIONAL: bool` generic argument, since we don't ever decide that based on a runtime value. Can leave it as is for simplicity, though

Reply:
Oh this is cool :D

Reply:
Took a stab at that in https://github.com/vectordotdev/vector/pull/14268/commits/a021406aaa8a3212ac407fd659822c6ed75d0936